### PR TITLE
Sprint23 EM v3 and microstructure replacing #267 and #270

### PR DESCRIPTION
**Pull request by mkuehbach**

### CODE REVIEW

File: contributed_definitions/nyaml/NXoptical_system_em.yaml [hunk 1]
```diff
@@ -7,9 +7,14 @@ NXoptical_system_em(NXobject):
   # NEW ISSUE: all the definitions here should better be backed up by the
   # work of the HMC EM glossary activities
   camera_length(NX_NUMBER):
-    doc: |
-      Citing the JEOL TEM glossary this is *an effective distance from a specimen
-      to a plane where an observed diffraction pattern is formed*.
+    doc:
+    - |
+      Distance which is present between the specimen surface and the detector plane.
```
Comment (lukaspie) [inline contributed_definitions/nyaml/NXoptical_system_em.yaml:12]:
```suggestion
      Distance between the specimen surface and the detector plane.
```

**Review by lukaspie — COMMENTED** [review]
just some small remaining comments, but nothing that should change functionality

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml [hunk 2]
```diff
@@ -88,138 +88,160 @@ doc: |
...
+        Edges are interpreted by inspecting the direction of their outer unit normals
+        (which point either parallel or antiparallel) along respective base vector direction
+        of the reference frame.
+        
+        If any of these assumptions is not met, the user is required to explicitly state this.
+      enumeration: [undefined, front_top_left, front_top_right, front_bottom_right, front_bottom_left, back_top_left, back_top_right, back_bottom_right, back_bottom_left]
+    x_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing x-axis base vector of the sample_reference_frame.
+      enumeration: [undefined, north, east, south, west, in, out]
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml:202]: Indeed listing explicit values for enumerations here is a constraint. Currently, I use the values in some of the EM examples given that these terms are agreed upon at least by the community users of specific software from the EM community (MTex here the case). I have removed them from this base class and move them to the application definition.

File: contributed_definitions/nyaml/NXcorrector_cs.yaml [hunk 3]
```diff
@@ -59,10 +63,108 @@ NXcorrector_cs(NXcomponent_em):
...
+    c_5_0(NXaberration):
+      # magnitude(NX_NUMBER):
+      #   unit: NX_LENGTH
+    c_5_2_a(NXaberration):
+    c_5_2_b(NXaberration):
+    c_5_4_a(NXaberration):
+    c_5_4_b(NXaberration):
+    c_5_6_a(NXaberration):
+    c_5_6_b(NXaberration):
+    # further comments from Thilo Remmele (Leibniz-Institut für Kristallzüchtung)
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcorrector_cs.yaml:129]: If this applies to the comments in lines L129 and following I agree, if it applies to the specifically named aberration coefficients - no these will not be removed as they are agreed upon terminology as referred to by model (ceos or nion convention)

[the remark this answers is not in the document]
Comment (lukaspie) [reply contributed_definitions/nyaml/NXcorrector_cs.yaml:129]: This is about the comments

File: contributed_definitions/nyaml/NXcrystal_structure.yaml [hunk 4]
```diff
@@ -129,8 +124,7 @@ NXcrystal_structure(NXobject):
     unit: NX_ANY
     \@depends_on(NX_CHAR):
       doc: |
-        Reference to an instance of :ref:`NXcoordinate_system`
-        whereby the positions can be resolved.
+        Details the reference frame in which the positions are defined.
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcrystal_structure.yaml:127]: I changed the docstring to remove the reference to NXcoordinate_system.

File: contributed_definitions/nyaml/NXem.yaml [hunk 5]
```diff
@@ -108,94 +109,104 @@ NXem(NXobject):
...
         into other files as one could write inference and information/fact/knowledge retrieval
         algorithm(s) whereby a specific piece of information that is related to an experiment
         or simulation is read directly from the respective file of the technology partner.
         
-        The reason why currently this works convincingly in hardly any research data
-        management system (RDMS) is the strong heterogeneity of the information contained
-        in such files and the fact that often context and documentation, specifically
-        rich semantic documentation, and contextualization is missing.
-      # using NXserialized here instead of NXnote as the former is more specific
+        The reason why currently this works convincingly in hardly any research data management system
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXem.yaml:137]: Good suggestion, I removed it here and placed it as part of the em docs.

File: contributed_definitions/nyaml/NXcorrector_cs.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXcorrector_cs.yaml:129]: Okay, then agreed, will be removed when Thilo has given feedback which he hasnt

File: contributed_definitions/nyaml/NXem.yaml [hunk 6]
```diff
@@ -108,94 +109,104 @@ NXem(NXobject):
...
+        doc: 
+        - |
+          Telephone number at the point in time when the experiment was performed.
       role(NX_CHAR):
-        doc: |
-          Which role does the user have in the place and at the point
-          in time when the experiment was performed? Technician operating
-          the microscope, student, postdoc, principle investigator, or guest
-          are common examples.
+        doc:
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXem.yaml:180]: Implemented

File: contributed_definitions/nyaml/NXem.yaml [hunk 7]
```diff
@@ -108,94 +109,104 @@ NXem(NXobject):
...
+        - |
+          User role at the point in time when the experiment was performed.
+        - |
+          Examples are technician operating the microscope, student, postdoc,
+          principle investigator, or guest.
       identifier(NXidentifier):
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXem.yaml:186]: for consistence with other fields here, yes why not

File: contributed_definitions/nyaml/NXem.yaml [hunk 8]
```diff
@@ -108,94 +109,104 @@ NXem(NXobject):
...
-      # user: -> not matched right now
-      # citation: doi ->why relevant, should be solved by RDMS
-      doc: |
-        A description of the material characterized in the experiment.
-        Sample and specimen are threaded as de facto synonyms.
-        Samples can be real specimens or virtual (see method).
+      doc:
+      - |
+        A physical entity which contains material intended to be investigated.
+        Sample and specimen are threaded as de facto synonyms. Samples can be real or virtual ones.
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXem.yaml:200]: good catch!

File: contributed_definitions/nyaml/NXem.yaml [hunk 9]
```diff
@@ -108,94 +109,104 @@ NXem(NXobject):
...
+      - |
+        xref:
+          spec: EMglossary
+          term: Specimen
+          url: https://purls.helmholtz-metadaten.de/emg/EMG_00000046
       method(NX_CHAR):
```
[the remark this answers is not in the document]
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXem.yaml:206]: Indeed type from the NIAC base class fits better, changed it

File: contributed_definitions/nyaml/NXoptical_system_em.yaml (hunk 1 above)
Comment (mkuehbach) [reply contributed_definitions/nyaml/NXoptical_system_em.yaml:12]: Rejected, using a community agreed upon definition, https://owl.emglossary.helmholtz-metadaten.de/index.html#EMG_00000008

File: contributed_definitions/nyaml/NXcoordinate_system_set.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lukaspie) [reply contributed_definitions/nyaml/NXcoordinate_system_set.yaml]: ok

**Review by lukaspie — COMMENTED** [review]
lgtm

**Review by lukaspie — APPROVED** [review]

**Pull request merged**